app/models/pet.rb

```diff
@@ -10,4 +12,8 @@ def shelter_name
   def self.adoptable
     where(adoptable: true)
   end
+
+  def get_pet_by_name(name)
+    Pet.where(name: name)
```

Comment:
Remember that every model method you write should be tested in a model spec! You could also do `find_by(name: xxx)` and do this in the controller